src/goto-programs/slice_global_inits.cpp

```diff
@@ -67,8 +67,12 @@ void slice_global_inits(goto_modelt &goto_model)
     {
       const codet &code = i_it->code;
       find_symbols(code, symbols, true, false);
-      const exprt &expr = i_it->guard;
-      find_symbols(expr, symbols, true, false);
+
+      if(i_it->has_condition())
+      {
+        const exprt &expr = i_it->get_condition();
```

Comment:
Maybe we can just get rid of this temporary that doesn't actually add value?

Reply:
done